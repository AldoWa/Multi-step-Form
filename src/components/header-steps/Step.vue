<template>
  <div class="w-9 h-9 rounded-full flex items-center justify-center"
    :class="isActualStep ? 'bg-[#FC6C4C]' : 'bg-[#E5E7EB]'"
    data-test="step"
  >
    <span :class="isActualStep ? 'text-white' : 'text-[#111827]'"
    >{{ step }}</span>
  </div>
</template>

<script setup lang="ts">
interface StepProps {
  step: number
  isActualStep: boolean;
}
defineProps<StepProps>()
</script>