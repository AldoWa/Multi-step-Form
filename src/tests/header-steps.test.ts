import { mount } from '@vue/test-utils'

import { it, describe, expect } from 'vitest'

import Step from '../components/header-steps/Step.vue'

describe('Header Steps', () => {
  describe('Step', () => {
    it('Should render the component', () => {
      const wrapper = mount(Step, {
        props: {
          step: 1,
          isActualStep: false,
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  
    it('Should render the component with the correct step', () => {
      const wrapper = mount(Step, {
        props: {
          step: 2,
          isActualStep: false,
        }
      })

      expect(wrapper.get('[data-test="step"]').text()).toBe('2')
    })

    it('Should render the actualStep', () => {
      const wrapper = mount(Step, {
        props: {
          step: 2,
          isActualStep: true,
        }
      })

      expect(wrapper.get('[data-test="step"]').classes()).toContain('bg-[#FC6C4C]')
    })
  })
})