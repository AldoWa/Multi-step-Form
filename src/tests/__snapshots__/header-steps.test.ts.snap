// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Steps > Step > Should render the component 1`] = `"<div class="w-9 h-9 rounded-full flex items-center justify-center bg-[#E5E7EB]" data-testid="step"><span class="text-[#111827]">1</span></div>"`;
